### PR TITLE
KS endpoint type lookup for Keystone, Cinder & Glance Charts

### PR DESCRIPTION
**Pull request by intlabs**
**What is the purpose of this pull request?**: To implement Keystone endpoint lookups by type for the Keystone & Cinder Charts

**What issue does this pull request address?**: Superceeds https://github.com/att-comdev/openstack-helm/pull/208

**Notes for reviewers to consider**:
This is part of a series of PR's that will change each chart to use the Endpoint Type lookup method over the Endpoint Name for Keystone Endpoints.

**Specific reviewers for pull request**:
@alanmeadows @v1k0d3n @wilkers-steve @larryrensing @gardlt

### CODE REVIEW

**Comment by intlabs** [comment]
@alanmeadows If you could review at this stage it would be really appreciated.

I am aware that the lookup function names are not ideal - but would like to change them as part of the cleanup of functions that these PR's will make redundant, in order to minimize impact the upon other charts.

File: cinder/templates/service-api.yaml [hunk 1]
```diff
@@ -15,9 +15,9 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ .Values.service.api.name }}
```
Comment (intlabs) [inline cinder/templates/service-api.yaml]: Though I would like the service name to be configurable, I removed that option here to be consistent with other charts.

Comment (v1k0d3n) [inline cinder/templates/service-api.yaml]: i think that's fine. we can address/discuss once we're in openstack.

**Comment by intlabs** [comment]
@v1k0d3n/@alanmeadows I'm fine with either merging this PR and then hitting the follow up charts in a series of rapid PRs before we move, or accumulating all of this work under this one. Though before progressing on either path I'd like to ensure that the work to date makes the grade :)

**Comment by v1k0d3n** [comment]
since @alanmeadows is out, @wilkers-steve and @larryrensing can you review this as well?

@intlabs as long as we have commitment to keep driving these child PR's (please submit an Issue, as a show of intent), then I'm fine with pulling this in as is...just need the commitment.

**Comment by intlabs** [comment]
@v1k0d3n, I'll see it through. Hopefully to the sweet, not bitter, end. Cheers :)

File: cinder/templates/service-api.yaml (hunk 1 above)
Comment (wilkers-steve) [inline cinder/templates/service-api.yaml]: Agreed here

**Comment by larryrensing** [comment]
looks good to me

**Comment by v1k0d3n** [comment]
@larryrensing or @wilkers-steve: i know you're getting a 500 when using the "Commits" tab, but can you review by selecting the "Files Changed" tab as a workaround? There are 17 files committed in this PR (at the moment), and we can still review and get merged. I think @intlabs and I are saying that things look good, but I'd like to get another set of eyes on it formally.

btw: an issue has been submitted with the github team. i'm sure that this is a bug related to their UI changes this past week.

**Comment by v1k0d3n** [comment]
@wilkers-steve strange, but thanks for checking it out. hopefully we don't run into this issue again. 

thanks @intlabs! nice cleanup.

**Pull request merged**